rotkehlchen/chain/arbitrum_one/modules/umami/decoder.py

```diff
@@ -0,0 +1,175 @@
+REWARD_TOPIC: Final = b'q\xba\xb6\\\xed.WPwZ\x06\x13\xbe\x06}\xf4\x8e\xf0l\xf9*In\xbfvc\xae\x06`\x92IT'  # noqa: E501
+DEPOSIT_EXECUTION_FOUR_BYTES = b'\xdb\x10\xc3\xb9'
+
+
+class UmamiDecoder(ArbitrumDecoderInterface):
+    def __init__(
+            self,
+            evm_inquirer: 'ArbitrumOneInquirer',
+            base_tools: 'BaseDecoderTools',
+            msg_aggregator: 'MessagesAggregator',
+    ) -> None:
+        super().__init__(
+            evm_inquirer=evm_inquirer,
+            base_tools=base_tools,
+            msg_aggregator=msg_aggregator,
+        )
+
+    def _decode_deposit_withdraw_request(self, context: DecoderContext, event: 'EvmEvent') -> None:
+        """Decode an event in a deposit/withdraw request."""
+        is_deposit = context.tx_log.topics[0] == DEPOSIT_TOPIC
+        raw_amount = int.from_bytes(context.tx_log.data[0:32] if is_deposit else context.tx_log.data[32:64])  # noqa: E501
+        event.counterparty = CPT_UMAMI
+
+        if event.balance.amount == (amount := asset_normalized_value(
+                amount=raw_amount,
+                asset=event.asset.resolve_to_crypto_asset(),
+        )):  # deposit or withdraw event - balance will match amount in log data
+            if is_deposit:
+                event.event_type = HistoryEventType.DEPOSIT
+                event.event_subtype = HistoryEventSubType.DEPOSIT_ASSET
+                event.notes = f'Deposit {amount} {event.asset.symbol_or_name()} into Umami'
+            else:
+                event.event_type = HistoryEventType.SPEND
+                event.event_subtype = HistoryEventSubType.RETURN_WRAPPED
+                event.notes = f'Return {amount} {event.asset.symbol_or_name()} to Umami'
+
+        elif event.balance.amount == asset_normalized_value(
+                amount=context.transaction.value,
+                asset=event.asset.resolve_to_crypto_asset(),
+        ):  # Execution cost, balance will match transaction value
+            event.event_type = HistoryEventType.DEPOSIT if is_deposit else HistoryEventType.WITHDRAWAL  # noqa: E501
+            event.event_subtype = HistoryEventSubType.FEE
+            event.notes = f'Execution cost of {event.balance.amount} ETH'
+
+    def _decode_umami_vault_events(self, context: DecoderContext) -> DecodingOutput:
+        """Decode umami vault deposit/withdraw events.
+
+        A deposit/withdraw consists of two transactions, a request tx and an execution tx.
+        The event for deposit execution is not available in decoded_events here,
+        and must be transformed later via an ActionItem.
+        """
+        found_relevant_events = False
+        for event in context.decoded_events:
+            if (
+                event.event_type == HistoryEventType.SPEND and
+                event.event_subtype == HistoryEventSubType.NONE and
+                context.tx_log.topics[0] in {DEPOSIT_TOPIC, WITHDRAW_TOPIC}
```

Comment:
for example here you are already checking and getting the topic type and then later in `._decode_deposit_withdraw_request` you again check against it multiple times. Best abstract it beter, check once and route it in the appropriate function somewhere around here